protobuf-3.6.1.cmake

```diff
@@ -27,35 +27,68 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-# https://tracker.debian.org/pkg/gdal
+# https://tracker.debian.org/pkg/protobuf
+
+set(version        3.12.3)
+set(download_hash  SHA256=71030a04aedf9f612d2991c1c552317038c3c5a2b578ac4745267a45e7037c29)
+set(patch_version  ${version}-2)
+set(patch_hash     SHA256=953cbd01d0e3aadb978658d939f68008ca249ce7fdba7c41267022fb7c872a0c)
+set(base_url       https://snapshot.debian.org/archive/debian/20200708T024852Z/pool/main/p/protobuf/)
 
-set(version        3.6.1)
-set(download_hash  SHA256=3d4e589d81b2006ca603c1ab712c9715a76227293032d05b26fca603f90b3f5b)
-set(patch_version  ${version})
-set(base_url       https://github.com/protocolbuffers/protobuf/archive/)
 
 superbuild_package(
-  NAME           protobuf
+  NAME           protobuf-patches
+  VERSION        ${patch_version}
+  
+  SOURCE
+    URL            ${base_url}protobuf_${patch_version}.debian.tar.xz
+    URL_HASH       ${patch_hash}
+)
+
+superbuild_package(
+  NAME           protobuf-source
   VERSION        ${patch_version}
   DEPENDS
-    common-licenses
-    zlib
+    source:protobuf-patches-${patch_version}
   
   SOURCE
-    URL            ${base_url}v${version}/protobuf-cpp-${version}.tar.gz
+    URL            ${base_url}protobuf_${version}.orig.tar.gz
     URL_HASH       ${download_hash}
+    PATCH_COMMAND
+      "${CMAKE_COMMAND}"
+        -Dpackage=protobuf-patches-${patch_version}
+        -P "${APPLY_PATCHES_SERIES}"
+)
+
+add_custom_target(protobuf-${patch_version}-source)
+set_property(TARGET protobuf-${patch_version}-source
+  PROPERTY SB_SOURCE_DIR "${PROJECT_BINARY_DIR}/source/protobuf-source-${patch_version}/cmake"
+)
```

Comment:
This is an undocumented way to handle a non-standard source directory. I use it for CI of OpenOrienteering Mapper (and CupCalculator), with Azure Pipeline's original git checkout. I should use it for libpolyclipping, too.

Reply:
Nice, it took me a while to figure out a workaround